src/sacn/source.c

```diff
@@ -220,22 +611,76 @@ size_t sacn_source_get_universes(sacn_source_t handle, uint16_t* universes, size
  * @return #kEtcPalErrInvalid: Invalid parameter provided.
  * @return #kEtcPalErrNotInit: Module not initialized.
  * @return #kEtcPalErrNotFound: Handle does not correspond to a valid source or the universe is not on that source.
+ * @return #kEtcPalErrExists: The unicast destination was already added to this universe on this source.
  * @return #kEtcPalErrSys: An internal library or system call error occurred.
  */
 etcpal_error_t sacn_source_add_unicast_destination(sacn_source_t handle, uint16_t universe, const EtcPalIpAddr* dest)
 {
-  ETCPAL_UNUSED_ARG(handle);
-  ETCPAL_UNUSED_ARG(universe);
-  ETCPAL_UNUSED_ARG(dest);
-  return kEtcPalErrNotImpl;
+#if !SACN_SOURCE_ENABLED
+  return kEtcPalErrNotInit;
+#endif
+
+  etcpal_error_t result = kEtcPalErrOk;
+
+  // Verify module initialized.
+  if (!sacn_initialized())
+    result = kEtcPalErrNotInit;
+
+  // Check for invalid arguments.
+  if (result == kEtcPalErrOk)
+  {
+    if (handle == SACN_SOURCE_INVALID)
```

Comment:
Combine these checks into a single if() block

Reply:
Resolved in https://github.com/ETCLabs/sACN/pull/32/commits/4338ffdded0288b92cc19437c16825145d3966f0.